bin/ubuntu-core-initramfs

```diff
@@ -486,6 +486,127 @@ def install_misc(dest_dir, sysroot):
     # FIXME: systemd is configured with the wrong path to dmsetup
     os.symlink("../usr/sbin/dmsetup", os.path.join(dest_dir, "sbin", "dmsetup"))
 
+
+class AptRepo:
+
+    def __init__(self, url):
+        self.url = url
+        self.suites = []
+        self.components = []
+
+
+def find_apt_repos(sysroot):
+    repos = {}
+    sources = os.path.join(sysroot, "etc/apt/sources.list")
+    sources_d = os.path.join(sysroot, "etc/apt/sources.list.d")
+    state_d = os.path.join(sysroot, "var/lib/apt/lists")
+    repo_lines = check_output(["apt-cache", "policy",
+                               "-o", "Dir::Etc::SourceList="+sources,
+                               "-o", "Dir::Etc::SourceParts="+sources_d,
+                               "-o", "Dir::State::Lists="+state_d]).decode("utf-8").splitlines()
+    uri_line = re.compile(r"^ *-?[0-9]+ http")
```

Comment:
Do not we have also ftp and rsync mirrors?

Reply:
Good point, I've added the possible transports as specified in `sources.list(5)`